File: src/styles/sidebar.cs
/* Sidebar Overlay */
.sidebar-overlay {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(0, 0, 0, 0.5);
  z-index: 998;
  display: none;
}

@media (max-width: 768px) {
  .sidebar-overlay {
    display: block;
  }
}

/* Main Sidebar Container */
.sidebar {
  position: fixed;
  top: 0;
  bottom: 0;
  width: 280px;
  background: #1f2937;
  color: white;
  z-index: 999;
  display: flex;
  flex-direction: column;
  transition: all 0.3s ease;
  transform: translateX(-100%);
  box-shadow: 2px 0 10px rgba(0, 0, 0, 0.1);
}

.sidebar.left {
  left: 0;
}

.sidebar.right {
  right: 0;
  transform: translateX(100%);
}

.sidebar.open {
  transform: translateX(0);
}

.sidebar.collapsed {
  width: 80px;
}

@media (min-width: 769px) {
  .sidebar {
    position: relative;
    transform: translateX(0);
  }
}

/* Sidebar Header */
.sidebar-header {
  padding: 1.5rem 1rem;
  border-bottom: 1px solid #374151;
  display: flex;
  align-items: center;
  justify-content: space-between;
  min-height: 80px;
  background: #111827;
}

.sidebar-logo {
  height: 2rem;
  width: auto;
}

.sidebar-title {
  font-size: 1.25rem;
  font-weight: bold;
  margin: 0;
  color: white;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.sidebar-toggle {
  background: #374151;
  border: none;
  color: white;
  width: 2rem;
  height: 2rem;
  border-radius: 0.25rem;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  transition: background 0.2s ease;
  font-size: 0.875rem;
}

.sidebar-toggle:hover {
  background: #4b5563;
}

.sidebar-toggle:focus {
  outline: 2px solid #3b82f6;
  outline-offset: 2px;
}

/* Navigation */
.sidebar-nav {
  flex: 1;
  padding: 1rem 0;
  overflow-y: auto;
  scrollbar-width: thin;
  scrollbar-color: #4b5563 #1f2937;
}

.sidebar-nav::-webkit-scrollbar {
  width: 6px;
}

.sidebar-nav::-webkit-scrollbar-track {
  background: #1f2937;
}

.sidebar-nav::-webkit-scrollbar-thumb {
  background: #4b5563;
  border-radius: 3px;
}

.sidebar-nav::-webkit-scrollbar-thumb:hover {
  background: #6b7280;
}

/* Menu Lists */
.sidebar-menu {
  list-style: none;
  margin: 0;
  padding: 0;
}

.sidebar-item,
.sidebar-group {
  margin-bottom: 0.25rem;
}

/* Menu Links and Group Headers */
.sidebar-link,
.sidebar-group-header {
  display: flex;
  align-items: center;
  padding: 0.75rem 1rem;
  color: #d1d5db;
  text-decoration: none;
  transition: all 0.2s ease;
  cursor: pointer;
  border: none;
  background: none;
  width: 100%;
  text-align: left;
  gap: 0.75rem;
  font-size: 0.875rem;
  font-weight: 500;
}

.sidebar-link:hover,
.sidebar-group-header:hover {
  background: #374151;
  color: white;
}

.sidebar-link.active,
.sidebar-group-header.active {
  background: #3b82f6;
  color: white;
  box-shadow: inset 3px 0 0 #1d4ed8;
}

.sidebar-link.disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.sidebar-link.disabled:hover {
  background: none;
  color: #d1d5db;
}

/* Icons */
.sidebar-icon {
  font-size: 1.25rem;
  width: 1.5rem;
  height: 1.5rem;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-shrink: 0;
}

/* Labels */
.sidebar-label {
  flex: 1;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

/* Badges */
.sidebar-badge {
  background: #ef4444;
  color: white;
  font-size: 0.75rem;
  padding: 0.125rem 0.375rem;
  border-radius: 9999px;
  min-width: 1.25rem;
  text-align: center;
  font-weight: 600;
  line-height: 1;
}

/* Expand Icon */
.expand-icon {
  font-size: 0.75rem;
  transition: transform 0.2s ease;
  color: #9ca3af;
}

.expand-icon.expanded {
  transform: rotate(180deg);
}

/* Submenu */
.sidebar-submenu {
  list-style: none;
  margin: 0;
  padding: 0;
  background: #111827;
  border-left: 2px solid #374151;
}

.sidebar-subitem {
  margin: 0;
}

.sidebar-sublink {
  display: flex;
  align-items: center;
  padding: 0.5rem 1rem 0.5rem 3rem;
  color: #9ca3af;
  text-decoration: none;
  transition: all 0.2s ease;
  gap: 0.75rem;
  font-size: 0.8125rem;
  font-weight: 400;
}

.sidebar-sublink:hover {
  background: #1f2937;
  color: white;
}

.sidebar-sublink.active {
  background: #3b82f6;
  color: white;
  box-shadow: inset 3px 0 0 #1d4ed8;
}

/* User Section */
.sidebar-user {
  padding: 1rem;
  border-top: 1px solid #374151;
  margin-top: auto;
  background: #111827;
}

.user-info {
  display: flex;
  align-items: center;
  gap: 0.75rem;
  margin-bottom: 0.75rem;
}

.user-avatar {
  width: 2.5rem;
  height: 2.5rem;
  border-radius: 50%;
  object-fit: cover;
  flex-shrink: 0;
  border: 2px solid #374151;
}

.user-details {
  flex: 1;
  min-width: 0;
}

.user-name {
  display: block;
  font-weight: 600;
  color: white;
  font-size: 0.875rem;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  line-height: 1.2;
}

.user-role {
  display: block;
  color: #9ca3af;
  font-size: 0.75rem;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  line-height: 1.2;
}

.logout-btn {
  width: 100%;
  padding: 0.5rem;
  background: #374151;
  color: white;
  border: none;
  border-radius: 0.25rem;
  cursor: pointer;
  font-size: 0.875rem;
  font-weight: 500;
  transition: background 0.2s ease;
}

.logout-btn:hover {
  background: #4b5563;
}

.logout-btn:focus {
  outline: 2px solid #3b82f6;
  outline-offset: 2px;
}

/* Footer */
.sidebar-footer {
  padding: 1rem;
  border-top: 1px solid #374151;
  text-align: center;
  background: #111827;
}

.sidebar-version {
  color: #6b7280;
  font-size: 0.75rem;
  font-weight: 400;
}

/* Collapsed State Styles */
.sidebar.collapsed .sidebar-title,
.sidebar.collapsed .sidebar-label,
.sidebar.collapsed .sidebar-badge,
.sidebar.collapsed .expand-icon,
.sidebar.collapsed .user-details,
.sidebar.collapsed .logout-btn,
.sidebar.collapsed .sidebar-version {
  display: none;
}

.sidebar.collapsed .sidebar-submenu {
  display: none;
}

.sidebar.collapsed .sidebar-link,
.sidebar.collapsed .sidebar-group-header {
  justify-content: center;
  padding: 0.75rem;
  position: relative;
}

.sidebar.collapsed .user-info {
  justify-content: center;
  margin-bottom: 0;
}

.sidebar.collapsed .sidebar-header {
  justify-content: center;
  padding: 1rem;
}

/* Tooltip for collapsed state */
.sidebar.collapsed .sidebar-link:hover::after,
.sidebar.collapsed .sidebar-group-header:hover::after {
  content: attr(title);
  position: absolute;
  left: 100%;
  top: 50%;
  transform: translateY(-50%);
  background: #1f2937;
  color: white;
  padding: 0.5rem 0.75rem;
  border-radius: 0.25rem;
  font-size: 0.875rem;
  white-space: nowrap;
  z-index: 1000;
  margin-left: 0.5rem;
  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
  border: 1px solid #374151;
}

/* Responsive Styles */
@media (max-width: 768px) {
  .sidebar {
    width: 280px;
  }
  
  .sidebar.collapsed {
    width: 280px;
  }
  
  .sidebar.collapsed .sidebar-title,
  .sidebar.collapsed .sidebar-label,
  .sidebar.collapsed .sidebar-badge,
  .sidebar.collapsed .expand-icon,
  .sidebar.collapsed .user-details,
  .sidebar.collapsed .logout-btn,
  .sidebar.collapsed .sidebar-version {
    display: block;
  }
  
  .sidebar.collapsed .sidebar-link,
  .sidebar.collapsed .sidebar-group-header {
    justify-content: flex-start;
    padding: 0.75rem 1rem;
  }
  
  .sidebar.collapsed .user-info {
    justify-content: flex-start;
    margin-bottom: 0.75rem;
  }
}

/* Animation for menu items */
@keyframes slideIn {
  from {
    opacity: 0;
    transform: translateX(-10px);
  }
  to {
    opacity: 1;
    transform: translateX(0);
  }
}

.sidebar-submenu {
  animation: slideIn 0.2s ease-out;
}

/* Focus styles for accessibility */
.sidebar-link:focus,
.sidebar-group-header:focus,
.sidebar-sublink:focus {
  outline: 2px solid #3b82f6;
  outline-offset: -2px;
}

/* High contrast mode support */
@media (prefers-contrast: high) {
  .sidebar {
    border-right: 2px solid #fff;
  }
  
  .sidebar-link.active,
  .sidebar-group-header.active {
    background: #000;
    color: #fff;
  }
}

File: src/components/common/Sidebar.js
import React, { useState } from 'react';
import '../styles/sidebar.css';

const Sidebar = ({
  isOpen = true,
  onToggle,
  logo,
  title = "Dashboard",
  menuItems = [],
  userProfile = null,
  onLogout,
  position = "left", // "left" or "right"
  collapsible = true,
  showUserSection = true,
  activeItem = null,
  onItemClick
}) => {
  const [expandedGroups, setExpandedGroups] = useState({});
  const [isCollapsed, setIsCollapsed] = useState(false);

  const toggleGroup = (groupId) => {
    setExpandedGroups(prev => ({
      ...prev,
      [groupId]: !prev[groupId]
    }));
  };

  const toggleCollapse = () => {
    setIsCollapsed(!isCollapsed);
    if (onToggle) onToggle(!isCollapsed);
  };

  const handleItemClick = (item, e) => {
    if (item.onClick) {
      item.onClick(e);
    }
    if (onItemClick) {
      onItemClick(item);
    }
  };

  const renderMenuItem = (item, index) => {
    const isActive = activeItem === item.id || activeItem === item.label;
    const hasChildren = item.children && item.children.length > 0;
    const isExpanded = expandedGroups[item.id] || expandedGroups[item.label];

    if (hasChildren) {
      return (
        <li key={index} className="sidebar-group">
          <button
            className={`sidebar-group-header ${isActive ? 'active' : ''}`}
            onClick={() => toggleGroup(item.id || item.label)}
          >
            {item.icon && <span className="sidebar-icon">{item.icon}</span>}
            {!isCollapsed && (
              <>
                <span className="sidebar-label">{item.label}</span>
                <span className={`expand-icon ${isExpanded ? 'expanded' : ''}`}>
                  ▼
                </span>
              </>
            )}
          </button>
          {isExpanded && !isCollapsed && (
            <ul className="sidebar-submenu">
              {item.children.map((child, childIndex) => (
                <li key={childIndex} className="sidebar-subitem">
                  <a
                    href={child.href || '#'}
                    className={`sidebar-sublink ${activeItem === child.id || activeItem === child.label ? 'active' : ''}`}
                    onClick={(e) => handleItemClick(child, e)}
                  >
                    {child.icon && <span className="sidebar-icon">{child.icon}</span>}
                    <span className="sidebar-label">{child.label}</span>
                    {child.badge && (
                      <span className="sidebar-badge">{child.badge}</span>
                    )}
                  </a>
                </li>
              ))}
            </ul>
          )}
        </li>
      );
    }

    return (
      <li key={index} className="sidebar-item">
        <a
          href={item.href || '#'}
          className={`sidebar-link ${isActive ? 'active' : ''} ${item.disabled ? 'disabled' : ''}`}
          onClick={(e) => !item.disabled && handleItemClick(item, e)}
        >
          {item.icon && <span className="sidebar-icon">{item.icon}</span>}
          {!isCollapsed && (
            <>
              <span className="sidebar-label">{item.label}</span>
              {item.badge && (
                <span className="sidebar-badge">{item.badge}</span>
              )}
            </>
          )}
        </a>
      </li>
    );
  };

  return (
    <>
      {/* Overlay for mobile */}
      {isOpen && <div className="sidebar-overlay" onClick={onToggle}></div>}
      
      <aside className={`sidebar ${position} ${isOpen ? 'open' : ''} ${isCollapsed ? 'collapsed' : ''}`}>
        {/* Header Section */}
        <div className="sidebar-header">
          {logo ? (
            <img src={logo} alt={title} className="sidebar-logo" />
          ) : (
            !isCollapsed && <h2 className="sidebar-title">{title}</h2>
          )}
          
          {collapsible && (
            <button
              className="sidebar-toggle"
              onClick={toggleCollapse}
              aria-label="Toggle sidebar"
            >
              {isCollapsed ? '→' : '←'}
            </button>
          )}
        </div>

        {/* Navigation Menu */}
        <nav className="sidebar-nav">
          <ul className="sidebar-menu">
            {menuItems.map((item, index) => renderMenuItem(item, index))}
          </ul>
        </nav>

        {/* User Section */}
        {showUserSection && userProfile && (
          <div className="sidebar-user">
            <div className="user-info">
              {userProfile.avatar && (
                <img 
                  src={userProfile.avatar} 
                  alt="User Avatar" 
                  className="user-avatar"
                />
              )}
              {!isCollapsed && (
                <div className="user-details">
                  <span className="user-name">{userProfile.name}</span>
                  <span className="user-role">{userProfile.role}</span>
                </div>
              )}
            </div>
            {!isCollapsed && onLogout && (
              <button
                className="logout-btn"
                onClick={onLogout}
                title="Logout"
              >
                Logout
              </button>
            )}
          </div>
        )}

        {/* Footer Section */}
        <div className="sidebar-footer">
          {!isCollapsed && (
            <div className="sidebar-version">
              v1.0.0
            </div>
          )}
        </div>
      </aside>
    </>
  );
};

// Default menu items example
Sidebar.defaultProps = {
  menuItems: [
    {
      id: 'dashboard',
      label: 'Dashboard',
      icon: '🏠',
      href: '/dashboard'
    },
    {
      id: 'analytics',
      label: 'Analytics',
      icon: '📊',
      children: [
        { id: 'reports', label: 'Reports', href: '/analytics/reports' },
        { id: 'metrics', label: 'Metrics', href: '/analytics/metrics' }
      ]
    },
    {
      id: 'users',
      label: 'Users',
      icon: '👥',
      href: '/users',
      badge: '12'
    },
    {
      id: 'settings',
      label: 'Settings',
      icon: '⚙️',
      href: '/settings'
    }
  ]
};

export default Sidebar;